sdk/billing/Azure.ResourceManager.Billing/api/Azure.ResourceManager.Billing.net8.0.cs

```diff
@@ -4561,7 +4563,10 @@ public BillingRoleAssignmentProperties(Azure.Core.ResourceIdentifier roleDefinit
         public Azure.Core.ResourceIdentifier BillingRequestId { get { throw null; } }
         public string CreatedByPrincipalId { get { throw null; } }
         public string CreatedByPrincipalPuid { get { throw null; } }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        [System.ObsoleteAttribute("This property is now deprecated. Please use the new property `CreatedByPrincipalTenantIdString` moving forward.")]
         public System.Guid? CreatedByPrincipalTenantId { get { throw null; } }
+        public string CreatedByPrincipalTenantIdString { get { throw null; } }
```

Comment:
this fix is quite a brutal force one - we should understand the issue first:
when do we have the format exception?
is the only case of getting a format exception because the service is returning the empty string indicating this is empty?
If this is the case, considering this property is readonly, instead of the current solution, we should override its deserialization, and check, if this is empty string, we deserialize this as null.

If the server returns other random strings which are not guid, the current solution is valid, but we really should polish the new property name.

Reply:
fixed.